### PR TITLE
nat: increase retries on collisions for newer kernels

### PR DESCRIPTION
**Pull request by borkmann**
See commit msg.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9294)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

File: bpf/lib/nat.h [hunk 1]
```diff
@@ -47,11 +47,18 @@ struct nat_entry {
 #define NAT_PUNT_TO_STACK	1
 
 #ifdef HAVE_LRU_MAP_TYPE
-#define NAT_MAP_TYPE BPF_MAP_TYPE_LRU_HASH
+# define NAT_MAP_TYPE BPF_MAP_TYPE_LRU_HASH
```
Comment (brb) [inline bpf/lib/nat.h:50]: Are these spaces intentional?

Comment (borkmann) [reply bpf/lib/nat.h:50]: Yes, it's typically done in order to denote that this define is under the given ifdef; basically kernel coding style.

**Review by brb — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by borkmann** [comment]
test-me-please

**Pull request merged**